EIPS/eip-signTypedData.md

```diff
@@ -0,0 +1,263 @@
+# eth_signTypedData
+
+## Preamble
+
+    EIP: <to be assigned>
+    Title: Ethereum typed data signing
+    Author: Leonid Logvinov <logvinov.leon@gmail.com>
+    Type: Standard Track
+    Category: ERC
+    Status: Draft
+    Created: 2017-09-13
+    Replaces: 693
+
+## Simple Summary
+
+Standard for machine-verifiable and human-readable typed data signing with Ethereum keys.
+
+## Abstract
+
+Ethereum clients provide the ability to sign UTF-8 strings with the `eth_sign` RPC call. This call however, doesn't give Signer UI's enough metadata to display the actual intent of the DApp when the data being signed is not a string. Over time, we expect value to be transferred on Ethereum using protocols that involve off-chain components (e.g state channels). These protocols are going to involve signing complex data structures with Ethereum keys, making a human-readable, machine-verifiable signing flow critical to user security.
+
+This EIP adds an RPC method to sign arrays of arbitrary typed data, making it easy for the Signer UI to display this data to the user in a human-readable form, so that the user can be 100% confident of exactly what they are signing, while keeping the resulting signatures machine-verifiable. In addition, there is no room for a malicious DApp to provide the signer with alternative data then that being presented to the user, since the signer hashes the typedData before the user signs it.
+
+This method should only be implemented in Ethereum clients with Signer UI's that require user approval before signing (MetaMask, MEW, Parity signer, Ledger, Trezor). It should not be implemented in clients where approval is already granted (e.g Geth with unlocked accounts).
+
+<img src="https://github.com/0xProject/EIPs/tree/master/EIPS/eip-eth_signTypedData/eth_signTypedData.png" width="500px">
```

Comment:
The image is not shown correctly in https://github.com/0xProject/EIPs/blob/01dfc0f9a4122d8ad8817c503447cab8efa8a6c4/EIPS/eip-signTypedData.md . Maybe `https://raw.githubusercontent.com/0xProject/EIPs/master/EIPS/eip-eth_signTypedData/eth_signTypedData.png` works.